src/utilities/EventEmitter.test.ts

```diff
@@ -0,0 +1,242 @@
+import { it, describe, expect, beforeEach, afterEach, beforeAll } from 'vitest';
+import EventEmitter, { removeListener } from './EventEmitter';
+
+describe('Removes a listener from an array or object of listeners', () => {
+  const listener = () => true;
+  const altListener = () => false;
+  
+  const targetListenerArrayFixture1 = [altListener, listener, altListener];
+  const targetListenerArrayFixture2 = [listener, altListener, listener];
+
+  it('Successfully removes a listener from arrays', () => {
+    const targetListeners = [[...targetListenerArrayFixture1], [...targetListenerArrayFixture2]];
+    removeListener(targetListeners, listener);
+    expect(targetListeners).toStrictEqual([[altListener, altListener], [altListener]]);
+  });
+  it('Successfully removes a listener from objects', () => {
+    const targetListeners = [{
+      'fixture1': [...targetListenerArrayFixture1],
+      'fixture2': [...targetListenerArrayFixture2],
+    }];
+    removeListener(targetListeners, listener);
+    expect(targetListeners).toStrictEqual([
+      {
+        'fixture1': [altListener, altListener],
+        'fixture2': [altListener]
+      }
+    ]);
+  });
+
+  it('Successfully handles a mixture of objects and arrays', () => {
+    const targetListeners = [{
+      'fixture1': [...targetListenerArrayFixture1],
+      'fixture2': [...targetListenerArrayFixture2],
+    }, [...targetListenerArrayFixture1], [...targetListenerArrayFixture2]];
+    removeListener(targetListeners, listener);
+    expect(targetListeners).toStrictEqual([
+      {
+        'fixture1': [altListener, altListener],
+        'fixture2': [altListener]
+      },
+      [altListener, altListener], [altListener]
+    ],)
+  });
+
+  it('Only removes listeners from a filtered list of events', () => {
+    const targetListeners = [{
+      'fixture1': [...targetListenerArrayFixture1],
+      'fixture2': [...targetListenerArrayFixture2],
+    }];
+    removeListener(targetListeners, listener, 'fixture2');
+    expect(targetListeners).toStrictEqual([
+      {
+        'fixture1': [altListener, listener, altListener],
+        'fixture2': [altListener]
+      }
+    ]);
+    removeListener(targetListeners, altListener, 'fixture1');
+    expect(targetListeners).toStrictEqual([
+      {
+        'fixture1': [listener],
+        'fixture2': [altListener]
+      }
+    ]);
+  })
+});
+
+describe('Event emitter class', () => {
+  let eventEmitter: EventEmitter<{ 'myEvent': string, 'myOtherEvent': string, 'myThirdEvent': string }>;
+  const listener = () => true;
+  const altListener = () => false;
+
+  afterEach(() => {
+    eventEmitter = new EventEmitter();
+  });
+
+  describe('calling the on method', () => {
+    beforeEach(() => {
+      eventEmitter = new EventEmitter();
```

Comment:
can this go into the vitest test context?

Reply:
Yes, it can